### PR TITLE
HPCC-20850 Implement ExpireDays attribute copying in DFU file copy.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Implement expireDays attribute copy.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested manually

### CODE REVIEW

**Comment by jakesmith** [comment]
@AttilaVamos - PR checkboxes need completing.

**Comment by AttilaVamos** [comment]
@jakesmith done.

File: dali/ft/filecopy.cpp [hunk 1]
```diff
@@ -3211,7 +3211,8 @@ void FileSprayer::updateTargetProperties()
                      ((stricmp(aname,"@blockCompressed")==0)&&copyCompressed) ||
                      ((stricmp(aname,"@rowCompressed")==0)&&copyCompressed)||
                      (stricmp(aname,"@local")==0)||
-                     (stricmp(aname,"@recordCount")==0)
+                     (stricmp(aname,"@recordCount")==0) ||
+                     (stricmp(aname,"@expireDays")==0)
```
Comment (jakesmith) [inline dali/ft/filecopy.cpp]: trivial: not new, but strieq would be cleaner.

Or could define the simple ones in a static array/vector/set and use find.

**Comment by jakesmith** [comment]
@AttilaVamos - looks fine as is.

**Comment by jakesmith** [comment]
@richardkchapman - ready to merge.

**Comment by jakesmith** [comment]
@AttilaVamos - so on 2nd thoughts, as per Richard's comments in meeting - it doesn't seem correct to automatically copy expireDays to copies of files.

There should be a way via FileServices to set expiry days (and remove them).

**Comment by AttilaVamos** [comment]
@richardkchapman @jakesmith yes, I can agree and I put this issue as a topic idea to our off-site agenda. I will close his PR. 
Perhaps we can keep the related JIRA for further discussion and/or implement other method to handle expiry days (and/or other logical file) attribute.

**Pull request closed without merging**